datalake/src/main/java/com/sequenceiq/datalake/repository/SdxStatusRepository.java

```diff
@@ -25,4 +27,8 @@
     List<SdxStatusEntity> findDistinctFirstByStatusInAndDatalakeIdInOrderByIdDesc(Collection<DatalakeStatusEnum> datalakeStatusEnums,
             Collection<Long> datalakeId);
 
+    @CheckPermission(action = ResourceAction.READ)
+    @Query("SELECT sdxstatus.datalake.id FROM SdxStatusEntity sdxstatus WHERE sdxstatus.status IN ('DELETED') GROUP BY sdxstatus.datalake.id")
+    Set<Long> findAllSdxClusterIdWhichHasDeletedState();
```

Comment:
It is just a workaround and we need to fix the original problem. When we set datalake status to deleted, then we update the deleted timestamp, but they are not in one transaction.